openmls/src/storage/kat_storage_stability.rs

```diff
@@ -0,0 +1,722 @@
+        &group_id,
+        &alice_group,
+    );
+
+    alice_group.merge_pending_commit(&alice_provider).unwrap();
+
+    let mut testdata_bob_added = vec![];
+    alice_provider
+        .storage
+        .serialize(&mut testdata_bob_added)
+        .unwrap();
+
+    alice_group
+        .update_group_context_extensions(
+            &alice_provider,
+            Extensions::single(Extension::RequiredCapabilities(
+                RequiredCapabilitiesExtension::new(&[ExtensionType::Unknown(0xf042)], &[], &[]),
+            )),
+            &alice_signer,
+        )
+        .unwrap();
+
+    let mut testdata_pending_gce_commit = vec![];
+    alice_provider
+        .storage
+        .serialize(&mut testdata_pending_gce_commit)
+        .unwrap();
+
+    check_serialized_group_equality::<_, Provider>(
+        &mut testdata_pending_gce_commit.as_slice(),
+        "alice",
+        &group_id,
+        &alice_group,
+    );
+
+    alice_group.merge_pending_commit(&alice_provider).unwrap();
+
+    let mut testdata_gce_updated = vec![];
+    alice_provider
+        .storage
+        .serialize(&mut testdata_gce_updated)
+        .unwrap();
+
+    check_serialized_group_equality::<_, Provider>(
+        &mut testdata_gce_updated.as_slice(),
+        "alice",
+        &group_id,
+        &alice_group,
+    );
+
+    //// also serialize with a pending proposal
+
+    let charlie_kpb = KeyPackageBuilder::new()
+        .leaf_node_capabilities(Capabilities::new(
+            None,
+            None,
+            Some(&[ExtensionType::Unknown(0xf042)]),
+            None,
+            None,
+        ))
+        .build(
+            ciphersuite,
+            &charlie_provider,
+            &charlie_signer,
+            charlie_cwk.clone(),
+        )
+        .unwrap();
+
+    alice_group
+        .propose_add_member(&alice_provider, &alice_signer, charlie_kpb.key_package())
+        .unwrap();
+
+    let mut testdata_pending_proposal = vec![];
+    alice_provider
+        .storage
+        .serialize(&mut testdata_pending_proposal)
+        .unwrap();
+
+    check_serialized_group_equality::<_, Provider>(
+        &mut testdata_pending_proposal.as_slice(),
+        "alice",
+        &group_id,
+        &alice_group,
+    );
+
+    (
+        group_id,
+        vec![
+            testdata_new_group,
+            testdata_pending_add_commit,
+            testdata_bob_added,
+            testdata_pending_gce_commit,
+            testdata_gce_updated,
+            testdata_pending_proposal,
+        ],
+    )
+}
+
+#[openmls_test]
+fn generate_kats(ciphersuite: Ciphersuite, provider: &Provider) {
+    helper_generate_kat::<Provider>(ciphersuite);
+}
+
+#[test]
+#[ignore]
+#[cfg(not(all(
+    feature = "libcrux-provider",
+    not(any(
+        target_arch = "wasm32",
+        all(target_arch = "x86", target_os = "windows")
+    ))
+)))]
+fn write_kats() {
+    // setup
+    let rustcrypto_provider = openmls_rust_crypto::OpenMlsRustCrypto::default();
```

Comment:
Except for the provider the `write_kats` functions look very similar. Can we factor out the core to avoid duplication?